backend/webui_service/webui_init.go

```diff
@@ -161,14 +163,15 @@ func (webui *WEBUI) Start() {
 	if factory.WebUIConfig.Configuration.Mode5G {
 		// get config file info from WebUIConfig
 		mongodb := factory.WebUIConfig.Configuration.Mongodb
-
-		// Connect to MongoDB
-		MongoDBLibrary.SetMongoDB(mongodb.Name, mongodb.Url)
+		// Create the MongoDB Client
+		var c, _ = mongoapi.NewMongoClient(mongodb.Url, mongodb.Name)
 		for {
-			if MongoDBLibrary.Client != nil {
+			if c.Client != nil {
+				commonDBClient.Client = c.Client
 				break
 			} else {
-				MongoDBLibrary.SetMongoDB(mongodb.Name, mongodb.Url)
+				c, _ = mongoapi.NewMongoClient(mongodb.Url, mongodb.Name)
```

Comment:
Should have some timer. Else this will be in tight loop.

Reply:
We add a timeout here.